chain/network/src/peer_manager/connection/mod.rs

```diff
@@ -182,6 +184,48 @@ impl Connection {
             }
         }
     }
+
+    pub fn send_snapshot_hosts(
```

Comment:
Can you please add some comments for this function?
What are `ds` and `sum`? Why does `sum` need to be computed here? Do you assume that `shard_ids` match exactly for the same peer/epoch_height ?